### PR TITLE
Simplify PR test pipeline (SUSE/avant-garde#1090)

### PR DESCRIPTION
**Pull request by Itxaka**
- Tests that need to run in pr are now marked pr
 - Pipeline for PR now only executes those marked tests
 - Steps removed from cluster deployment to make it simpler
 - Cleanup of makefile of unused test targets

## Why is this PR needed?

Does it fix an issue? addresses a business case?

add a description and link to the issue if one exists.

Fixes https://github.com/SUSE/avant-garde/issues/1090

**Reminder**: Add the "fixes bsc#XXXX" to the title of the commit so that it will
appear in the changelog.

## What does this PR do?

please include a brief "management" technical overview (details are in the code)

## Anything else a reviewer needs to know?

Special test cases, manual steps, links to resources or anything else that could be helpful to the reviewer.

## Info for QA

This is info for QA so that they can validate this. This is **mandatory** if this PR fixes a bug.
If this is a new feature, a good description in "What does this PR do" may be enough.

### Related info

Info that can be relevant for QA:
* link to other PRs that should be merged together
* link to packages that should be released together
* upstream issues

### Status **BEFORE** applying the patch

How can we reproduce the issue? How can we see this issue? Please provide the steps and the prove
this issue is not fixed.

### Status **AFTER** applying the patch

How can we validate this issue is fixed? Please provide the steps and the prove this issue is fixed.

## Docs

If docs need to be updated, please add a link to a PR to https://github.com/SUSE/doc-caasp.
At the time of creating the issue, this PR can be work in progress (set its title to [WIP]),
but the documentation needs to be finalized before the PR can be merged. 

# Merge restrictions

(Please do not edit this)

We are in *v4-maintenance phase*, so we will restrict what can be merged to prevent unexpected surprises:

    What can be merged (merge criteria):
        2 approvals:
            1 developer: code is fine
            1 QA: QA is fine
        there is a PR for updating documentation (or a statement that this is not needed)

### CODE REVIEW

**Comment by Itxaka** [comment]
> test_pre_bootstrap and test_post_bootstrap were not running anything, as the filter result was an empty set, right?

You are correct, they didnt really do anything

> now the tests that will be part of the PR need the new marker pr right? I like it, I dislike negation use for filtering as it was before, then it is not obvious or quick to get the result set of the filter.

Indeed, mark as PR, it wil run on a PR as one would expect :)

File: ci/infra/testrunner/kubectl/kubectl.py [hunk 1]
```diff
@@ -20,3 +20,19 @@ def run_kubectl(self, command, stdin=None):
             return self.utils.runshellcommand(shell_cmd, stdin=stdin)
         except Exception as ex:
             raise Exception("Error executing cmd {}".format(shell_cmd)) from ex
+
+    def get_node_names_by_role(self, role):
```
Comment (pablochacin) [inline ci/infra/testrunner/kubectl/kubectl.py]: Not sure what this change has to do with the purpose of this PR.

File: ci/infra/testrunner/testrunner.py [hunk 2]
```diff
@@ -70,11 +70,18 @@ def join_node(options):
...
-        timeout=options.timeout
-    )
+    try:
+        skuba.join_nodes(
+            masters=options.masters,
+            workers=options.workers,
+            timeout=options.timeout
+        )
+    except Exception:
+        skuba.join_nodes(
```
Comment (pablochacin) [inline ci/infra/testrunner/testrunner.py]: Not sure what this change has to do with the purpose of this PR. But in any case, looks like in case of an exception joining nodes, you are forcing to execute it again? Am I missing or misreading this change?

Comment (Itxaka) [reply ci/infra/testrunner/testrunner.py]: Sorry, this is a mistake on my part. This is not passing the vmware tests once it passed them once, so I already removed the WIP tag but I shouldnt as I been using this PR as test bed for some WIP actions to remediate vmware issues.

**Comment by davidcassany** [comment]
I think it would also make sense to remove the skuba unit tests from this pipeline and move it to a new one. I like the idea of having clearly separated the unit tests and the testrunner tests in the PR view. Also I think that pipelines should be simple and not mix different types of tests. Unit tests are way different different of what testrunner is actually doing.

Probably something that we could do in a follow up PR, but this is still related on PRs pipeline simplification. 

@Itxaka @pablochacin what you think?

**Comment by pablochacin** [comment]
What worries me about including the unit tests is that I'm not sure about their reliability, as they had not been part of the CI since some time ago.  

I vote for not including now and discuss in a separate ticket if include them later in this pipeline or create a separate job. I don't have a strong opinion on either.

**Comment by pablochacin** [comment]
What worries me about including the unit tests is that I'm not sure about their reliability, as they had not been part of the CI since some time ago.  

I vote for not including now and discuss in a separate ticket if include them later in this pipeline or create a separate job. I don't have a strong opinion on either.

**Comment by davidcassany** [comment]
> What worries me about including the unit tests is that I'm not sure about their reliability, as they had not been part of the CI since some time ago.

huh, I am confused now. The unit tests are already part of this pipeline, I am in favor of removing them from the pipeline and just keep testrunner tests here.

Are you aiming for removing them from the pipeline now and open a card to decide were to put those tests (new pipe or add them to an existing one)? I like the idea

**Comment by Itxaka** [comment]
I would move to move them into their own job (required), dont care if in this ticket on in another one.

> What worries me about including the unit tests is that I'm not sure about their reliability, as they had not been part of the CI since some time ago.

Arent they run on the test-pr and test-vmware-pr pipelines each time as a step?

**Comment by Itxaka** [comment]
> Are you aiming for removing them from the pipeline now and open a card to decide were to put those tests (new pipe or add them to an existing one)? I like the idea

I think we should not move them out if we dont have the substitution ready as that means not running working tests and potentially letting some errors in skuba slip in. Maybe some corner cases not covered by e2e tests..

**Comment by Itxaka** [comment]
> Are you aiming for removing them from the pipeline now and open a card to decide were to put those tests (new pipe or add them to an existing one)? I like the idea

I think we should not move them out if we dont have the substitution ready as that means not running working tests and potentially letting some errors in skuba slip in. Maybe some corner cases not covered by e2e tests..

**Comment by Itxaka** [comment]
> Are you aiming for removing them from the pipeline now and open a card to decide were to put those tests (new pipe or add them to an existing one)? I like the idea

I think we should not move them out if we dont have the substitution ready as that means not running working tests and potentially letting some errors in skuba slip in. Maybe some corner cases not covered by e2e tests..

**Comment by davidcassany** [comment]
> Arent they run on the test-pr and test-vmware-pr pipelines each time as a step?

Probably I am getting it wrong but those lines https://github.com/SUSE/skuba/blob/46dfb6e8e8d49df7fc3acb614d5d80cec1eb19d6/ci/jenkins/pipelines/prs/skuba-test.Jenkinsfile#L85-L89 suggest me that unit tests are executed just before running testrunner.

We could just discuss it in a follow up card for a second round of simplification.

**Comment by Itxaka** [comment]
> We could just discuss it in a follow up card for a second round of simplification.

I would say that it deserves a different card as its a different thing to do. Is not really simplifying, rather extracting a test into its own job.

**Comment by Itxaka** [comment]
test_remove_worker -> failing consistently -> New test, it migth have not worked before
test_hard_reboot[master-1] -> failing consistently -> existing test, what has changed?
test_hard_reboot[worker-0] -> failing consistently -> existing test, what has changed?

**Comment by pablochacin** [comment]
Maybe I didn't explain correctly. The job for unit tests exists, but I'm not sure tests are being executed at all:
https://ci.suse.de/job/caasp-jobs/job/v4/job/pr/job/update-unit/view/change-requests/job/PR-963/

Screenshot from 2020-03-03 10-37-37

Maybe is because the execution is conditional:
```
stage('skuba-update SUSE Unit Tests') {
            when {
                expression {
                    sh(script: "skuba/${PR_MANAGER} filter-pr --filename ${FILTER_SUBDIRECTORY}", returnStdout: true, label: "Filtering PR") =~ "contains changes"
                }
            }
            steps {
                dir("skuba/skuba-update") {
                    sh(script: "make test", label: 'skuba-update Unit Tests')
                }
            }
        }
```

**Comment by davidcassany** [comment]
> Maybe I didn't explain correctly. The job for unit tests exists, but I'm not sure tests are being executed at all:
> https://ci.suse.de/job/caasp-jobs/job/v4/job/pr/job/update-unit/view/change-requests/job/PR-963/

Well those tests are for `skuba-update` and only executed on `skuba-update` changes, unrelated to `skuba` tool.

**Comment by pablochacin** [comment]
> > Maybe I didn't explain correctly. The job for unit tests exists, but I'm not sure tests are being executed at all:
> > https://ci.suse.de/job/caasp-jobs/job/v4/job/pr/job/update-unit/view/change-requests/job/PR-963/
> 
> Well those tests are for `skuba-update` and only executed on `skuba-update` changes, unrelated to `skuba` tool.

You are right.
